resources/regexes.ts

```diff
@@ -1,7 +1,11 @@
+import { BaseRegExp } from '../types/trigger';
+
 export type Params<T extends string> =
-  Partial<Record<T, string | string[]> &
+  Partial<Record<Exclude<T, 'timestamp' | 'capture'>, string | string[]> &
   { 'timestamp': string; 'capture': boolean }>;
 
+export type Return<T extends string> = BaseRegExp<Exclude<T, 'capture'>>;
```

Comment:
I do have one naming question here, in that I understand that this is `Return` because it's what that the regex/netregex functions return.  However, I think it *is* a regex.  Maybe it could be called `Regex` for clarity? (Maybe it'd be worth having `Regex` and `NetRegex` differently typed versions too so that you can't mix and match them?)

Reply:
OK, that is not difficult.